lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Dictionary.java

```diff
@@ -1025,6 +1025,7 @@ private int writeNormalizedWordEntry(StringBuilder reuse, ByteSequencesWriter wr
     assert morphSep > 0;
     assert morphSep > flagSep;
     int sep = flagSep < 0 ? morphSep : flagSep;
+    if (sep == 0) return 0;
```

Comment:
We don't store empty dictionary entries anymore: they bring no benefits, only trouble.